x-pack/plugins/aiops/public/components/log_categorization/log_categorization_page.tsx

```diff
@@ -0,0 +1,327 @@
+import { useCategorizeRequest } from './use_categorize_request';
+import type { EventRate, Category, SparkLinesPerCategory } from './use_categorize_request';
+import { CategoryTable } from './category_table';
+import { DocumentCountChart } from './document_count_chart';
+
+export interface LogCategorizationPageProps {
+  dataView: DataView;
+  savedSearch: SavedSearch | SavedSearchSavedObject | null;
+}
+
+const BAR_TARGET = 20;
+
+export const LogCategorizationPage: FC<LogCategorizationPageProps> = ({
+  dataView,
+  savedSearch,
+}) => {
+  const {
+    notifications: { toasts },
+  } = useAiopsAppContext();
+
+  const { runCategorizeRequest, cancelRequest } = useCategorizeRequest();
+  const [aiopsListState, setAiopsListState] = useState(restorableDefaults);
+  const [globalState, setGlobalState] = useUrlState('_g');
+  const [selectedField, setSelectedField] = useState<string | undefined>();
+  const [selectedCategory, setSelectedCategory] = useState<Category | null>(null);
+  const [categories, setCategories] = useState<Category[] | null>(null);
+  const [currentSavedSearch, setCurrentSavedSearch] = useState(savedSearch);
+  const [loading, setLoading] = useState(false);
+  const [totalCount, setTotalCount] = useState(0);
+  const [eventRate, setEventRate] = useState<EventRate>([]);
+  const [pinnedCategory, setPinnedCategory] = useState<Category | null>(null);
+  const [sparkLines, setSparkLines] = useState<SparkLinesPerCategory>({});
+
+  useEffect(() => {
+    return () => {
+      cancelRequest();
+    };
+  }, [cancelRequest]);
+
+  const setSearchParams = useCallback(
+    (searchParams: {
+      searchQuery: Query['query'];
+      searchString: Query['query'];
+      queryLanguage: SearchQueryLanguage;
+      filters: Filter[];
+    }) => {
+      // When the user loads saved search and then clear or modify the query
+      // we should remove the saved search and replace it with the index pattern id
+      if (currentSavedSearch !== null) {
+        setCurrentSavedSearch(null);
+      }
+
+      setAiopsListState({
+        ...aiopsListState,
+        searchQuery: searchParams.searchQuery,
+        searchString: searchParams.searchString,
+        searchQueryLanguage: searchParams.queryLanguage,
+        filters: searchParams.filters,
+      });
+    },
+    [currentSavedSearch, aiopsListState, setAiopsListState]
+  );
+
+  const {
+    documentStats,
+    timefilter,
+    earliest,
+    latest,
+    searchQueryLanguage,
+    searchString,
+    searchQuery,
+    intervalMs,
+  } = useData(
+    { currentDataView: dataView, currentSavedSearch },
+    aiopsListState,
+    setGlobalState,
+    undefined,
+    BAR_TARGET
+  );
+
+  useEffect(() => {
+    if (globalState?.time !== undefined) {
+      timefilter.setTime({
+        from: globalState.time.from,
+        to: globalState.time.to,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(globalState?.time), timefilter]);
```

Comment:
I recall this approach caused some issues in the ML plugin in the past. Instead, the [DatePickerWrapper](https://github.com/elastic/kibana/blob/52febd8807dbdab5632ca763f1e82546c65dc1c9/x-pack/plugins/aiops/public/components/date_picker_wrapper/date_picker_wrapper.tsx#L56) component in aiops should do [similar stuff](https://github.com/elastic/kibana/blob/76fcb3fab99ab28ab1c1938fe767db70d04478ba/x-pack/plugins/ml/public/application/components/navigation_menu/date_picker_wrapper/date_picker_wrapper.tsx#L87) as in the ML plugin

Reply:
This will affect the Explain Log Rate Spike code too, so can be done in a later refactoring PR. I've added it to the [improvements issue](https://github.com/elastic/kibana/issues/140715).